sale_quotation_number/models/sale_order.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# © 2010-2012 Andy Lu <andy.lu@elico-corp.com> (Elico Corp)
+# © 2013 Agile Business Group sagl (<http://www.agilebg.com>)
+# © 2017 valentin vinagre  <valentin.vinagre@qubiq.es> (QubiQ)
+# © 2020 Manuel Regidor  <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models, api
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.model
+    def create(self, vals):
+        company = self.env['res.company']._company_default_get('sale.order')
+        if not company.keep_name_so:
+            vals['name'] = self.env['ir.sequence'].next_by_code(
+                'sale.quotation') or '/'
+        return super(SaleOrder, self).create(vals)
+
+    @api.multi
+    def copy(self, default=None):
+        self.ensure_one()
+        if default is None:
+            default = {}
+        default['name'] = '/'
+        if self.origin and self.origin != '':
+            default['origin'] = self.origin + ', ' + self.name
+        else:
+            default['origin'] = self.name
+        return super(SaleOrder, self).copy(default)
+
+    @api.multi
+    def action_confirm(self):
+        if any(self.mapped("company_id.keep_name_so")):
+            super(SaleOrder, self.filtered(
+                lambda x: x.company_id.keep_name_so)).action_confirm()
+        for order in self.filtered(lambda x: not x.company_id.keep_name_so):
+            # Check next number but don't increase the sequence
+            sequence_id = self.env['ir.sequence'].search([
+                ('code', '=', 'sale.order'),
+                ('company_id', 'in', [order.company_id.id, False])
+            ], order='company_id')
+            if order.origin and order.origin != '':
+                quo = order.origin + ', ' + order.name
+            else:
+                quo = order.name
+            new_name = sequence_id.get_next_char(sequence_id.number_next_actual)
+            # save old values
+            old_origin = order.origin
+            old_name = order.name
+            order.write({
+                'origin': quo,
+                'name': new_name
+            })
+            if not super(SaleOrder, order).action_confirm():
```

Comment:
@manuelregidor Action confirm always returns True, does it not?  If it fails, it always does a raise, so this seems like a lot of overhead?